lib/MahoCLI/Commands/SysEncryptionKeyRegenerate.php

```diff
@@ -0,0 +1,226 @@
+        $output->writeln('');
+
+        /** @var QuestionHelper $questionHelper */
+        $questionHelper = $this->getHelper('question');
+        $question = new ConfirmationQuestion('Are you sure you want to continue? (y/N) ', false);
+        if (!$questionHelper->ask($input, $output, $question)) {
+            $output->writeln('<info>Operation cancelled.</info>');
+            return Command::SUCCESS;
+        }
+
+        $this->initMaho();
+
+        // Disable config cache for current run
+        Mage::app()->getCacheInstance()->banUse('config');
+        Mage::app()->getConfig()->reinit();
+
+        $oldKey = $this->oldEncryptionKey = Mage::getEncryptionKeyAsHex();
+        $newKey = $this->newEncryptionKey = Mage::generateEncryptionKeyAsHex();
+        $currentDate = date('Y-m-d');
+        $localXmlPath = 'app/etc/local.xml';
+        $backupPath = 'app/etc/local.xml.bak.' . $currentDate;
+
+        // If it's an M1 encryption key check for mcrypt_compat
+        if (strlen($oldKey) !== SODIUM_CRYPTO_SECRETBOX_KEYBYTES * 2) {
+            $this->isOldEncryptionKeyM1 = true;
+            $output->writeln('');
+            $output->writeln('<error>It seems your encryption key is an old M1 one.</error>');
+            if (\Composer\InstalledVersions::isInstalled('phpseclib/mcrypt_compat')) {
+                $output->writeln('<error>Since you have mcrypt_compat installed we will try to re-encrypt all your crypted data.</error>');
+            } else {
+                $output->writeln('<error>Since you do not have mcrypt_compat installed we will not be able to re-encrypt your crypted data.</error>');
+            }
+
+            $output->writeln('');
+            $question = new ConfirmationQuestion('Are you sure you want to continue? (y/N) ', false);
+            if (!$questionHelper->ask($input, $output, $question)) {
+                $output->writeln('<info>Operation cancelled.</info>');
+                return Command::SUCCESS;
+            }
+        }
+
+        // Check if local.xml exists
+        if (!file_exists($localXmlPath)) {
+            $output->writeln('<error>Configuration file app/etc/local.xml not found</error>');
+            return Command::FAILURE;
+        }
+
+        // Create backup of local.xml
+        if (!copy($localXmlPath, $backupPath)) {
+            $output->writeln('<error>Failed to create backup file: ' . $backupPath . '</error>');
+            return Command::FAILURE;
+        }
+        $output->writeln('<info>Created backup at: ' . $backupPath . '</info>');
+
+        // Read the current configuration file and replace the encryption key
+        $localXmlContent = file_get_contents($localXmlPath);
+        $updatedContent = preg_replace(
+            '/<key(?:\s+date="[^"]*")?><!\[CDATA\[(.*?)\]\]><\/key>/',
+            '<key date="' . $currentDate . '"><![CDATA[' . $newKey . ']]></key>',
+            $localXmlContent,
+        );
```

Comment:
```suggestion
        $localXmlContent = simplexml_load_file($localXmlPath);
        $localXmlContent->global->crypt->key = $newKey;
        $localXmlContent->global->crypt->key->addAttribute('date', '');
        $localXmlContent->global->crypt->key->attributes()->date = $currentDate;

        $updatedContent = $localXmlContent->asXML();
```

I haven't fully tested this yet, but I'm always reminded of [this stackoverflow answer](https://stackoverflow.com/a/1732454) when parsing XML with regex.

I have a vague recollection that I've removed the `<![CDATA[` tag in my local.xml before because it's actually pointless here. The key will never contain characters that could be mistaken for XML.

Reply:
yep but it's what's already used in the install module for the rest of the things, that's why I did it that way, dunno if I would rewrite all of that in the context of this PR

about the CDATA, what if the password contains a "<" char, which I think it could be pretty normal?

Reply:
The CDATA would need to stay for the database password for sure. I just am not sure why Mage devs felt the need to add it around every other value.

But, If that's how the install module is now, then no reason to change anything in this PR, let's just leave well enough alone. :)